### ISSUE TITLE
Cannot work on neovim with lazyvim

### ISSUE THREAD

**Issue by Dav1dch** [BoringStuffInc/pylsp-inlay-hints#2]
Cannot work on neovim with lazyvim, can you show the configuration of how to enable the inlay hints in lazyvim?

**Comment by claytharrison** [comment]
Yes, it would be nice if someone could confirm that this works for them! I'm trying in emacs but `(eglot-server-capable :inlayHintProvider)` returns `nil` even when this is installed in the same environment as `python-lsp-server`. I don't really know where to start debugging. The README makes it sound like all you need to do is have the package installed, but I also tried enabling the plugin in my pylsp setup and that doesn't change anything.

**Comment by claytharrison** [comment]
OK so the only thing wrong, at least from the eglot perspective, is that the capability isn't properly exposed by pylsp the way eglot expects it. From what I can tell, the main issue is that pylsp only exposes inlayHintProvider as an *experimental* capability, rather than a proper top-level capability, which means eglot doesn't find it. I assume the issue is the same in neovim.

I don't know if there's any way for a pylsp plugin to add a new capability that isn't "experimental" (I can't find one). But basically just by modifying pylsp to say that it can be an inlayHintProvider at the top level, this plugin works just fine:

 https://github.com/python-lsp/python-lsp-server/issues/397#issuecomment-2558440324_
 
 Not sure if the right fix would be for pylsp to add a proper inlayHintProvider capability or for eglot/lazyvim/etc to scan for capabilities in the "experimental" section, but either of those should do it I guess.

**Comment by smheidrich** [comment]
@claytharrison I can confirm that Neovim does seem to ignore `experimental` capabilities as well and that your `python-lsp-server` patch from the linked comment (thank you for that, by the way!) makes it work.

I've opened a discussion at https://github.com/neovim/neovim/discussions/33865 to see whether Neovim can be configured to not ignore these.
